### DISCUSSION TITLE
YoloV8 deploy

### DISCUSSION THREAD

**Discussion by skkd113** [ultralytics/ultralytics#3973] [Q&A]
Hi, currently have a model trained, with an GUI using customtkinter. Everything working fine in Pycharm.
But when i try deploy to exe with auto-py-to-exe, this issue happens. Every time my software uses the function from yolo, my GUI be duplicated.
it just calls another GUI and nothing happens after that.

Sorry for my bad english

```
from ultralytics import YOLO
import time
import os
import customtkinter
current_path = os.path.dirname(os.path.abspath(__file__))
data_path = os.path.join(current_path, "data")
model = YOLO(os.path.join(current_path, "data","best.pt"))
def load_model():
    print("Loading model...")
    results = model.predict(os.path.join(current_path,"docs","rune_screenshots","dragon.png"),show=True)
    print("Done")   
if __name__ == "__main__":
    customtkinter.set_appearance_mode("light")  # Modes: system (default), light, dark
    customtkinter.set_default_color_theme("green")  # Themes: blue (default), dark-blue, green
    window = customtkinter.CTk()
    window.title("Test")
    window.geometry('50x50')
    start_button = customtkinter.CTkButton(window, width=80, height=26, text="Start", command=load_model)
    start_button.grid(row=0, column=1, padx=1, pady=1)
    window.mainloop()
```

**Comment by skkd113** [comment]
i tried without GUI but still bugged
```
from ultralytics import YOLO
import time
import os
import kthread
import customtkinter
import cv2
current_path = os.path.dirname(os.path.abspath(__file__))
data_path = os.path.join(current_path, "data")

model = YOLO(os.path.join(current_path, "data","best.pt"))
def load_model():
    print("Loading model...")
    results = model.predict(os.path.join(current_path,"docs","rune_screenshots","dragon.png"),show=True)
    cv2.waitKey(0)
    print("Done")
print("A")
load_model()
```
When i run the exe file, my project keep looping 

https://imgur.com/a/ry0mtNf

**Reply by glenn-jocher** [reply]
@skkd113 hello! It seems like you're encountering an issue where your GUI is being duplicated when you call the YOLO model's predict function after converting your script to an executable. This behavior is not typical and might be related to how the executable is handling multiprocessing or threading, which is often used within deep learning model inference functions.

When you're converting your script to an executable, it's important to ensure that all dependencies and environment settings are correctly packaged. The `auto-py-to-exe` tool should handle this, but there can be complications, especially with complex libraries that have non-Python dependencies or that spawn subprocesses.

Here are a few steps you can take to troubleshoot the issue:

1. **Check for Multiprocessing Issues**: Ensure that any multiprocessing code within the YOLO library is compatible with being frozen into an executable. Sometimes, multiprocessing can behave differently when a script is compiled.

2. **Review the Executable Build Logs**: Look for any warnings or errors that occur during the build process. These can provide clues as to what might be going wrong.

3. **Simplify Your Script**: Try to create the smallest script that reproduces the issue. This can sometimes make it easier to identify where things are going wrong.

4. **Test in Different Environments**: Try running the executable on a different machine or a clean Python environment to see if the issue persists. This can help determine if the problem is environment-specific.

5. **Consult the `auto-py-to-exe` Documentation**: There might be specific flags or settings you can use to better control the behavior of the executable.

6. **Check for Conflicts**: Ensure there are no conflicts between the GUI library and the YOLO library when packaged together.

If you continue to experience issues, consider reaching out on forums or communities specific to `auto-py-to-exe` or the libraries you're using for additional support. Remember that creating an executable from a Python script can introduce complexities, especially with advanced libraries like those used for machine learning.

For more detailed information on using YOLOv8, you can refer to our documentation at https://docs.ultralytics.com. Good luck with your deployment!
